apps/docs/src/pages/components/dialog.mdx

```diff
@@ -0,0 +1,376 @@
+import {
+  dialogClasses,
+  HvButton,
+  HvDialog,
+  HvDialogActions,
+  HvDialogContent,
+  HvDialogTitle,
+  theme,
+} from "@hitachivantara/uikit-react-core";
+
+import Playground from "@docs/components/code/Playground";
+import { Description } from "@docs/components/page/Description";
+import { Page } from "@docs/components/page/Page";
+import { getComponentData } from "@docs/utils/component";
+
+export const getStaticProps = async ({ params }) => {
+  const meta = await getComponentData("Dialog", "core", dialogClasses);
+  return { props: { ssg: { meta: meta } } };
+};
+
+<Description />
+
+<Page>
+
+### Maximum width
+
+Use the `maxWidth` prop to set the maximum width of the dialog. The width of the dialog will adjust automatically to the width of the content, unless the prop `fullWidth` is set to `true` in which case the dialog will grow to occupy the width specified on `maxWidth`.
+
+```tsx live
+import { useState } from "react";
+
+export default function Demo() {
+  const [open, setOpen] = useState(false);
+  const [maxWidth, setMaxWidth] = useState("md");
+  const [fullWidth, setFullWidth] = useState(true);
+
+  return (
+    <>
+      <HvButton style={{ width: "120px" }} onClick={() => setOpen(true)}>
+        Open dialog
+      </HvButton>
+      <HvDialog
+        maxWidth={maxWidth}
+        fullWidth={fullWidth}
+        open={open}
+        onClose={() => setOpen(false)}
+        style={{ overflow: "visible" }}
+      >
+        <HvDialogTitle variant="warning">Set Maximum Width</HvDialogTitle>
+        <HvDialogContent style={{ overflow: "visible" }} indentContent>
+          <SizeSelector
+            maxWidth={maxWidth}
+            fullWidth={fullWidth}
+            setMaxWidth={setMaxWidth}
+            setFullWidth={setFullWidth}
+          />
+        </HvDialogContent>
+        <HvDialogActions>
+          <HvButton
+            autoFocus
+            variant="secondaryGhost"
+            onClick={() => setOpen(false)}
+          >
+            Close
+          </HvButton>
+        </HvDialogActions>
+      </HvDialog>
+    </>
+  );
+}
+
+const SizeSelector = ({ maxWidth, setMaxWidth, fullWidth, setFullWidth }) => (
+  <div className="flex items-end gap-2 w-full justify-center">
```

Comment:
We should add the types here, but I think it's easier to just inline this instead of having a separate component.
These styles can be moved to `HvDialogContent`, simplifying things a bit